signer/utils.go

```diff
@@ -0,0 +1,109 @@
+package signer
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+)
+
+// DeriveSignatureAlgorithm picks up a recommended signing algorithm for given certificate.
+func DeriveSignatureAlgorithm(signingCert *x509.Certificate) (SignatureAlgorithm, error) {
```

Comment:
Nit, should we rename this to GetSignatureAlgorithm and GetKeySpec respectively?